src/components/GoogleAddressLookup/component.js

```diff
@@ -0,0 +1,297 @@
+                searchParams = Object.assign(searchParams, {
+                    bounds: new window.google.maps.LatLngBounds(
+                        new window.google.maps.LatLng(swLatitude, swLongitude),
+                        new window.google.maps.LatLng(neLatitude, neLongitude),
+                    ),
+                });
+            }
+
+            if (location) {
+                searchParams = Object.assign(searchParams, {
+                    location: new window.google.maps.LatLng(location.latitude, location.longitude),
+                });
+            }
+
+            if (country) {
+                searchParams = Object.assign(searchParams, {
+                    componentRestrictions: {
+                        country,
+                    },
+                });
+            }
+
+            searchParams = Object.assign(searchParams, otherSearchOptions);
+        }
+        this.setState({ isSearching: true });
+
+        return new Promise((resolve, reject) => {
+            this.autocompleteService.getPlacePredictions(
+                {
+                    ...searchParams,
+                    input: needle,
+                },
+                (predictions, status) => {
+                    this.setState({ isSearching: false });
+                    if (status !== window.google.maps.places.PlacesServiceStatus.OK) {
+                        return reject(status);
+                    }
+                    return resolve(predictions);
+                },
+            );
+        });
+    }
+
+    processSearchResults(results, searchValue) {
+        this.setState({
+            places: results,
+            suggestions: getSuggestions(results, searchValue),
+        });
+    }
+
+    handleError(searchValue) {
+        this.setState({
+            places: [],
+            suggestions: [getSearchForOption(searchValue)],
+        });
+    }
+
+    clear() {
+        this.setState({
+            places: [],
+            suggestions: [],
+        });
+    }
+
+    render() {
+        const {
+            style,
+            label,
+            error,
+            placeholder,
+            disabled,
+            tabIndex,
+            required,
+            id,
+            name,
+            value,
+            hideLabel,
+            onClick,
+            onFocus,
+            onBlur,
+        } = this.props;
+        const { isSearching, suggestions } = this.state;
+        const options = suggestions.length > 0 ? suggestions : null;
+        return (
+            <div id={id} style={style} className={this.getContainerClassName()}>
+                <Lookup
+                    id={this.lookupId}
+                    name={name}
+                    label={label}
+                    debounce
+                    isLoading={isSearching}
+                    placeholder={placeholder}
+                    options={options}
+                    hideLabel={hideLabel}
+                    readOnly={false}
```

Comment:
use:
```readOnly={readOnly}```
and add readOnly to index.js prop types in order to the user can use this prop